src/ooni/ooni_test_impl.hpp

```diff
@@ -174,36 +205,33 @@ class OoniTestImpl : public mk::NetTest {
         return new InputFileGenerator(input_filepath, logger);
     }
 
-    /*!
-     * \brief Start iterating over the input.
-     * \param cb Callback called when we are done.
-     */
-    virtual void begin(std::function<void()> cb) override {
-        geoip_lookup();
-        open_report();
-        if (input_filepath != "") {
-            logger->debug("net_test: found input file");
-            if (input != nullptr) {
-                delete input;
-                input = nullptr;
+    void begin(Callback<> cb) override {
```

Comment:
Here I removed the redundant `virtual` keyword, doxygen comment, and modernized the signature of the function using the more compact `Callback<>` alias rather than `std::function<>`.